### DISCUSSION TITLE
Total Least Squares (Solved)

### DISCUSSION THREAD

**Discussion by bas314** [casadi/casadi#3073] [General]
Hello all, 

I want to solve a total least squares problem to fit my data. I have error in the input as well as the output measurements. 
I try to solve this with Casadi, and my approach is to i) find the minimal distance to the function for given parameters, ii) minimise the sum of distances by changing the parameters. 

However, I get error messages that I do not understand. The code is below, and the errors below that. It fails on the last line.

Thanks for any help 
Bas

```
import casadi as ca

# the parameters to optimise
a = ca.MX.sym('a', 1)
b = ca.MX.sym('b', 1)

# I have F = a*r + b*r**3. I specify the value 'r' symbolically:
r = ca.MX.sym('r', 1)

# the measurements (only one now)
r_m = ca.MX.sym('r_m', 1)
F_m = ca.MX.sym('F_m', 1)

# given a measurement, minimal distance in input and output
Fhat = a*r + b*r**3
dr = (r - r_m)**2
dF = (Fhat - F_m)**2

# find the r (and F) to minimise distance, for given a, b, r_m, F_m
opt = {'x': r, 'f': dr + dF, 'p': ca.vertcat(a, b, r_m, F_m)}
solver = ca.nlpsol('min_dist', 'ipopt', opt)

# make an expression that only returns the distance
min_dist = solver(x0=[0.25], p=ca.vertcat(a, b, r_m, F_m))['f']

# optimise a, b
opt = {'x': ca.vertcat(a, b), 'f': min_dist, 'p': ca.vertcat(r_m, F_m) }
solver = ca.nlpsol('min_param', 'ipopt',
```
D:\bld\casadi_1657871574851\work\casadi\core\function_internal.cpp:145: Error calling IpoptInterface::init for 'min_param':
Error in Function::factory for 'nlp' [MXFunction] at D:\bld\casadi_1657871574851\work\casadi\core\function.cpp:1634:
Failed to create nlp_hess_l:[x, p, lam:f, lam:g]->[hess:gamma:x:x] with {"gamma": [f, g]}:
d:\bld\casadi_1657871574851\work\casadi\core\factory.hpp:387: Hessian generation failed:
Error in MX::hessian at D:\bld\casadi_1657871574851\work\casadi\core\mx.cpp:1679:
Error in MX::jacobian at D:\bld\casadi_1657871574851\work\casadi\core\mx.cpp:1663:
Error in XFunction::jac for 'helper_jacobian_MX' [MXFunction] at d:\bld\casadi_1657871574851\work\casadi\core\x_function.hpp:719:
Error in MXFunction::ad_forward at D:\bld\casadi_1657871574851\work\casadi\core\mx_function.cpp:831:
Error in MX::ad_forward for node of type class casadi::Call at D:\bld\casadi_1657871574851\work\casadi\core\mx.cpp:2035:
Error in Call::ad_forward for 'min_dist' [IpoptInterface] at D:\bld\casadi_1657871574851\work\casadi\core\casadi_call.cpp:123:
Error in Function::forward for 'min_dist' [IpoptInterface] at D:\bld\casadi_1657871574851\work\casadi\core\function.cpp:1017:
D:\bld\casadi_1657871574851\work\casadi\core\mx.cpp:521: Dimension mismatch for (x?y:0), x is 0x1, while y is 0x2

**Comment by bas314** [comment]
I copied the last line incomplete. It should be:

solver = ca.nlpsol('min_param', 'ipopt', opt)

**Comment by jgillis** [comment]
I can reproduce this error in 3.5.5, but not in 3.6.0 - can you check with 3.6.0 and let us know if it works correctly?
